internal/oci/container.go

```diff
@@ -527,17 +532,25 @@ func (c *Container) pid() (int, error) {
 // This is the simplest way to verify we are operating on the container
 // process, and haven't run into PID wrap.
 func (c *Container) verifyPid() error {
-	startTime, err := getPidStartTime(c.state.InitPid)
+	data, err := getPidData(c.state.InitPid)
 	if err != nil {
 		return err
 	}
 
-	if startTime != c.state.InitStartTime {
+	if data.StartTime != c.state.InitStartTime {
 		return fmt.Errorf(
 			"PID %d is running but has start time of %s, whereas the saved start time is %s. PID wrap may have occurred",
-			c.state.InitPid, startTime, c.state.InitStartTime,
+			c.state.InitPid, data.StartTime, c.state.InitStartTime,
 		)
 	}
+
+	if data.State == "D" {
```

Comment:
We could skip states that are considered invalid per:

```go
// Skip states which are typically considered as invalid.
switch data.State {
case "D", "X", "Z":
	return fmt.Errorf("PID %d has an invalid state: %s", c.state.InitPid, data.State)
}
```

Reply:
fixed. thanks